adapters/tlsio_mbedtls.c

```diff
@@ -565,35 +593,35 @@ int tlsio_mbedtls_close(CONCRETE_IO_HANDLE tls_io, ON_IO_CLOSE_COMPLETE on_io_cl
             tls_io_instance->tlsio_state = TLSIO_STATE_CLOSING;
             tls_io_instance->on_io_close_complete = on_io_close_complete;
             tls_io_instance->on_io_close_complete_context = callback_context;
-            mbedtls_ssl_close_notify(&tls_io_instance->ssl);
-
             if (xio_close(tls_io_instance->socket_io,
-                on_underlying_io_close_complete_during_close, tls_io_instance) != 0)
+                          on_underlying_io_close_complete_during_close, tls_io_instance) != 0)
             {
                 result = __FAILURE__;
             }
             else
             {
-                result = 0;
+                if (tls_io_instance->tls_status == 1)
```

Comment:
>if (tls_io_instance->tls_status == 1) [](start = 16, length = 37)

Please keep the check for other statuses (set return to 0 explicitly). Also, add a check for invalid status values and return __FAILURE__

Reply:
fixed